src/app/features/auth/thunks.ts

```diff
@@ -0,0 +1,50 @@
+import {
+    loginWithEmailPassword,
+    logoutFirebase,
+    registerUserWithEmailAndPassword,
+} from '@/firebase/providers'
+
+import { Dispatch } from '@reduxjs/toolkit'
+
+import { checkingCredentials, login, logout } from './authSlice'
+
+export const checkingAuthentication = () => {
+    return async (dispatch: Dispatch) => {
+        dispatch(checkingCredentials())
+    }
+}
+
+export const startRegisterWithEmailAndPassword = (
+    email: string,
+    password: string,
+    displayName: string
+) => {
+    return async (dispatch: Dispatch) => {
+        dispatch(checkingCredentials())
+
+        const { uid, ok, errorMessage } =
```

Comment:
```suggestion
        const { uid, ok: approved, errorMessage } =
```
This way we can rename the `ok` variable to a more clear name